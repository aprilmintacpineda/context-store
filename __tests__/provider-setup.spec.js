import React from 'react';
import renderer from 'react-test-renderer';
import Provider from '../lib';

const store = {
  todos: [
    {
      name: 'Code',
      status: 'Doing'
    }
  ]
};

test('Matches snapshot when persist prop was **NOT** provided', () => {
  expect(
    renderer
    .create(
      <Provider store={store}>
        <div>
          <h1>This is the app</h1>
        </div>
      </Provider>
    )
    .toJSON()
  ).toMatchSnapshot();
});

test('Matches snapshot when persist prop was provided', () => {
  const persist = {
    storage: window.localStorage,
    statesToPersist: storedStore => {
      return {
        todos: storedStore.todos
      };
    }
  };

  expect(
    renderer
    .create(
      <Provider store={store} persist={persist}>
        <div>
          <h1>This is the app</h1>
        </div>
      </Provider>
    )
    .toJSON()
  ).toMatchSnapshot();
});