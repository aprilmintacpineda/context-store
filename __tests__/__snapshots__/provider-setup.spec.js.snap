// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Matches snapshot when persist prop was **NOT** provided 1`] = `
<div>
  <h1>
    This is the app
  </h1>
</div>
`;

exports[`Matches snapshot when persist prop was provided 1`] = `
<div>
  <h1>
    This is the app
  </h1>
</div>
`;
